remote/terraform_state_file.go

```diff
@@ -44,10 +45,18 @@ func (state *TerraformState) IsRemote() bool {
 	return state.Remote != nil
 }
 
-// Parse the Terraform .tfstate file from its default locations. If the file doesn't exist at any of the default
-// locations, return nil.
-func ParseTerraformStateFileFromDefaultLocations() (*TerraformState, error) {
-	if util.FileExists(DEFAULT_PATH_TO_LOCAL_STATE_FILE) {
+// Parse the Terraform .tfstate file from the specified locations. If no location is specified,
+// search the default locations. If the file doesn't exist at any of the default locations, return nil.
+func ParseTerraformStateFileFromLocation(workingDir string) (*TerraformState, error) {
+	if workingDir != "" && util.FileExists(workingDir) {
```

Comment:
I don't think this check is necessary. If `workingDir` is empty, then `path.Join(workingDir, foo)` will just return `foo`. Moreover, `workingDir` should always be set, so that's not a particularly big concern.

In short, you can reduce this code back to the original if/else-if/else structure, but the paths in the if and else-if should both be `Join`ed with `workingDir`.